src/components/Planning/PlanningToolbar/PlanningToolbar.tsx

```diff
@@ -1,48 +1,95 @@
-import { Icon, IconButton, Toolbar } from '../../shared';
-import { ReactComponent as IconNewItem } from '@/assets/icons/new-item.svg';
-import { Navigation } from 'hds-react/components/Navigation';
-import { useTranslation } from 'react-i18next';
+import { Toolbar } from '../../shared';
+import { IconCollapse, IconPlusCircle, IconSort } from 'hds-react/icons/';
+import { useCallback, MouseEvent as ReactMouseEvent, useState, FC } from 'react';
+import { IPlanningRowSelections } from '@/interfaces/common';
+import { dispatchContextMenuEvent } from '@/utils/events';
+import { ContextMenuType } from '@/interfaces/eventInterfaces';
+import { Button } from 'hds-react/components/Button';
 import { GroupDialog } from '../GroupDialog';
 import { ProjectProgrammedDialog } from '../ProjectProgrammedDialog';
 import './styles.css';
-import { IconCross } from 'hds-react/icons/';
-import { useCallback, useRef } from 'react';
-import { IPlanningRowSelections } from '@/interfaces/common';
+import { t } from 'i18next';
 
-interface ProjectToolBarProps {
+interface IPlanningToolbarProps {
+  toggleGroupsExpanded: () => void;
+  groupsExpanded: boolean;
   selections: IPlanningRowSelections;
 }
-const ProjectToolbar = ({ selections }: ProjectToolBarProps) => {
-  const { Dropdown, Item } = Navigation;
-  const dropDownRef = useRef<HTMLDivElement>(null);
-  const closeDropdown = useCallback(() => {
-    dropDownRef.current?.click();
-  }, [dropDownRef]);
-  const { t } = useTranslation();
+const ProjectToolbar: FC<IPlanningToolbarProps> = ({
```

Comment:
memoize the component because it is receiving properties

Reply:
Done